module/documents/active-effect.mjs

```diff
@@ -399,24 +475,83 @@ export default class ActiveEffect5e extends ActiveEffect {
     return `${path}-${level}.${ext}`;
   }
 
+  /**
+   * Map the duration of an item to an active effect duration.
+   * @param {Item5e} item     An item with a duration.
+   * @returns {object}        The active effect duration.
+   */
+  static _getEffectDuration(item) {
+    const dur = item.system.duration ?? {};
+    const value = dur.value || 1;
+
+    switch (dur.units) {
+      case "turn": return { turns: value };
+      case "round": return { rounds: value };
+      case "minute": return { seconds: value * 60 };
+      case "hour": return { seconds: value * 60 * 60 };
+      case "day": return { seconds: value * 60 * 60 * 24 };
+      case "year": return { seconds: value * 60 * 60 * 24 * 365 };
+      default: return { seconds: 60 };
+    }
+  }
+
   /* -------------------------------------------- */
 
   /**
-   * Implement custom exhaustion cycling when interacting with the Token HUD.
+   * Implement custom exhaustion cycling when interacting with the Token HUD,
+   * and management of which items the actor is concentrating on.
    * @param {PointerEvent} event        The triggering event.
    */
-  static onClickTokenHUD(event) {
+  static async onClickTokenHUD(event) {
     const { target } = event;
-    if ( !target.classList?.contains("effect-control") || (target.dataset?.statusId !== "exhaustion") ) return;
+    if ( !target.classList?.contains("effect-control") ) return;
+
     const actor = canvas.hud.token.object?.actor;
-    let level = foundry.utils.getProperty(actor ?? {}, "system.attributes.exhaustion");
-    if ( !Number.isFinite(level) ) return;
-    event.preventDefault();
-    event.stopPropagation();
-    if ( event.button === 0 ) level++;
-    else level--;
-    const max = CONFIG.DND5E.conditionTypes.exhaustion.levels;
-    actor.update({ "system.attributes.exhaustion": Math.clamped(level, 0, max) });
+    if ( !actor ) return;
+    if ( target.dataset?.statusId === "exhaustion" ) {
+      let level = foundry.utils.getProperty(actor, "system.attributes.exhaustion");
+      if ( !Number.isFinite(level) ) return;
+      event.preventDefault();
+      event.stopPropagation();
+      if ( event.button === 0 ) level++;
+      else level--;
+      const max = CONFIG.DND5E.conditionTypes.exhaustion.levels;
+      actor.update({ "system.attributes.exhaustion": Math.clamped(level, 0, max) });
+    } else if ( target.dataset?.statusId === "concentrating" ) {
+      const effects = actor.concentration?.effects ?? new Set();
+      if ( effects.size < 1 ) return;
+      event.preventDefault();
+      event.stopPropagation();
+      if ( effects.size === 1 ) {
+        effects.first().delete();
+        return;
+      }
+      const choices = effects.reduce((acc, effect) => {
```

Comment:
I think it's clever, but I'm not sure how I feel about using a dialog that is historically used only for item *usage* for just a prompt to delete an effect.

I suspect this can cause a lot of confusion for modules relying on it.